requirements.txt

```diff
@@ -1,4 +1,5 @@
 beautifulsoup4>=4.5.1
+codecov
```

Comment:
I think the version for codecov should also be provided. What say @cclauss @AnshulMalik

Reply:
Pinning requirements to a particular version or better is a good idea but the more that I think about this PR...

We should not put testing tools into requirements.txt because we do not need them in production deployments.  Instead we should pip install these tools (flake8, pytest, codeassy, codecov, coverage) in .travis.yml.

I think it is better to leave requirements.txt as is and edit .travis.yml instead.

Reply:
@cclauss , if we don't include them in requirements.txt, then contributors will face problems.

We should provide the codecov version here :)

Reply:
pytest and flake8 are used by developers :(

Reply:
So is Codecov ??

Reply:
On my local machine I run flake8 and pytest on each PR but I do not run Codacy, codecov, and coverage.  __What test tools do you run on your local systems before submitting a PR?__

When we deploy to Heroku or Bluemix, I do not believe that any of Codacy, codecov, coverage, flake8, and pytest should be deployed for speed and RAM footprint reasons.  Given that the Python buildpacks that Heroku and Bluemix use will read in and pip install the module in requirements.txt.

Reply:
We should leave them as it is for now, if they cause peoblems or take too much space, then we can come up with some good solution for this.